numpy/core/shape_base.py

```diff
@@ -275,3 +278,67 @@ def hstack(tup):
         return _nx.concatenate(arrs, 0)
     else:
         return _nx.concatenate(arrs, 1)
+
+def stack(arrays, axis=0):
+    """
+    Join a sequence of arrays along a new axis.
```

Comment:
Although it is nice to indicate appending the axis with a simple `-1`. In the current case, an explanation of how negative axis values are handled would help.  For the list version you  could do something like
```
In [26]: a = ones((2,2,3))

In [27]: newshape = list(a.shape)

In [28]: newshape.insert(-1, 1)

In [29]: a.reshape(newshape).shape
Out[29]: (2, 2, 1, 3)
```